File: app/src/main/java/com/boardgamegeek/provider/GamesIdExpansionsIdProvider.java
package com.boardgamegeek.provider;

import android.content.ContentUris;
import android.net.Uri;

import com.boardgamegeek.provider.BggContract.GamesExpansions;
import com.boardgamegeek.util.SelectionBuilder;

public class GamesIdExpansionsIdProvider extends BaseProvider {
	final GamesIdExpansionsProvider provider = new GamesIdExpansionsProvider();

	@Override
	protected SelectionBuilder buildSimpleSelection(Uri uri) {
		long expansionId = ContentUris.parseId(uri);
		return provider.buildSimpleSelection(uri).whereEquals(GamesExpansions.EXPANSION_ID, expansionId);
	}

	@Override
	protected String getPath() {
		return addIdToPath(provider.getPath());
	}

	@Override
	protected String getType(Uri uri) {
		return GamesExpansions.CONTENT_ITEM_TYPE;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/GamesIdExpansionsIdProvider.kt
package com.boardgamegeek.provider

import android.content.ContentUris
import android.net.Uri
import com.boardgamegeek.provider.BggContract.Companion.PATH_EXPANSIONS
import com.boardgamegeek.provider.BggContract.Companion.PATH_GAMES
import com.boardgamegeek.provider.BggContract.Games
import com.boardgamegeek.provider.BggContract.GamesExpansions
import com.boardgamegeek.provider.BggDatabase.Tables

class GamesIdExpansionsIdProvider : BaseProvider() {
    override fun getType(uri: Uri) = GamesExpansions.CONTENT_ITEM_TYPE

    override val path = "$PATH_GAMES/#/$PATH_EXPANSIONS/#"

    override fun buildSimpleSelection(uri: Uri): SelectionBuilder {
        val gameId = Games.getGameId(uri)
        val expansionId = ContentUris.parseId(uri)
        return SelectionBuilder()
            .table(Tables.GAMES_EXPANSIONS)
            .whereEquals(GamesExpansions.Columns.GAME_ID, gameId)
            .whereEquals(GamesExpansions.Columns.EXPANSION_ID, expansionId)
    }
}
